prow/hook/events.go

```diff
@@ -430,10 +432,22 @@ func (s *Server) handleGenericComment(l *logrus.Entry, ce *github.GenericComment
 			)
 			start := time.Now()
 			labels := prometheus.Labels{"event_type": l.Data[eventTypeField].(string), "action": string(ce.Action), "plugin": p}
-			if err := h(agent, *ce); err != nil {
+			if err := func() (err error) {
+				defer func() {
+					if r := recover(); r != nil {
+						err = fmt.Errorf("plugin '%v' paniced with %v. stack is : \n%s", p, r, debug.Stack())
+					}
+				}()
+				err = h(agent, *ce)
```

Comment:
Nit: You can just `return err` here

Reply:
And yeah, given that we have a big bunch of handlers it might make sense to build a reusable func for this that itself takes a `func() error`

Reply:
I think `events.go` is fine